test/integration/targets/assemble/tasks/main.yml

```
@@ -29,16 +29,46 @@
     that:
     - "result.changed == true"
 
-- name: test assemble with all fragments
+- name: test assemble with all fragments, but only check!
   assemble: src="{{remote_tmp_dir}}/src" dest="{{remote_tmp_dir}}/assembled1"
+  check_mode: True
   register: result
 
 - name: assert the fragments were assembled
   assert:
     that:
-    - "result.state == 'file'"
     - "result.changed == True"
-    - "result.checksum == '74152e9224f774191bc0bedf460d35de86ad90e6'"
+
+- name: ensure file was not created
+  stat:
+    path: "{{remote_tmp_dir}}/assembled1"
+  register: check_it
+
+- name: it should not exist, yet
+  assert:
+    that:
+      - not check_it.stat.exists
+
+- name: test assemble with all fragments, really create now!
+  assemble: src="{{remote_tmp_dir}}/src" dest="{{remote_tmp_dir}}/assembled1"
```

I know this is copy-pasted but I would stay away from k=v syntax in newly added tasks?
```suggestion
  assemble:
    src: "{{ remote_tmp_dir }}/src"
    dest: "{{ remote_tmp_dir }}/assembled1"
```